language/move-vm/runtime/src/interpreter.rs

```diff
@@ -351,7 +357,13 @@ impl Interpreter {
         ty: &Type,
     ) -> PartialVMResult<AbstractMemorySize<GasCarrier>> {
         let gv = Self::load_resource(data_store, addr, ty)?;
-        let mem_size = gv.size();
+        let mem_size = gv.size().map(|mem_size| {
+            if mem_size == 0 {
+                MIN_EXISTS_DATA_SIZE.get()
+            } else {
+                mem_size
+            }
+        });
```

Comment:
cc @vgao1996 you've put a lot of good thought in here, so want to get your take on this to see if you think it sounds good? Would you prefer/have a different suggestion for this instead of having a base charge?

Reply:
Shouldn't we just fix `GlobalValue/GlobalValueImpl::size`?

Reply:
I would be fine with doing that. My only reason for not changing it here as opposed to there is that in this context it is a specific case where an "empty value" should have a non-zero size, whereas in `GlobablValue/GlobalValueImpl::size` this is the size of the value in a context-independent setting, so returning a non-zero size for the "empty value" seems incorrect/contradictory. Happy to do it either way, just let me know which you prefer.

Reply:
That's an interesting point. Although I'd say the empty value isn't really empty -- the new `GlobalValue` is more like a slot, and the empty state still holds a flag indicating whether the value is fresh or deleted. Anyways you certainly understand the gas semantics better than I so I'll trust your judgement on this.

Reply:
ahh very good point, I hadn't realized that about `GlobalValue`. I think I'll go with your recommendation here and move this in to the `size` impl for `GlobalValue/GlobalValueImpl` since that will much cleaner as well.